hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -423,7 +423,11 @@ protected void preWrite(String instantTime, WriteOperationType writeOperationTyp
       HoodieTableMetaClient metaClient) {
     setOperationType(writeOperationType);
     this.lastCompletedTxnAndMetadata = TransactionUtils.getLastCompletedTxnInstantAndMetadata(metaClient);
-    this.asyncCleanerService = AsyncCleanerService.startAsyncCleaningIfEnabled(this);
+    if (null == this.asyncCleanerService) {
```

Comment:
this changes behavior? currently the async cleaner is started for each write, seems like we are reusing this now? Is this intentional. Can we retain existing behavior?

Reply:
> currently the async cleaner is started for each write

This behavior is kept for new code.

> Is this intentional

Yes, instantiate a new thread pool is not a good design and is risky for leak, we better not do that.